manifests/params.pp

```diff
@@ -16,98 +16,71 @@
 #
 class openvpn::params {
 
-  case $::osfamily {
-    'RedHat': {
+  case $facts['os']['family'] {
+    'RedHat': { # RedHat/CentOS
       $etc_directory       = '/etc'
       $root_group          = 'root'
       $group               = 'nobody'
       $link_openssl_cnf    = true
       $pam_module_path     = '/usr/lib64/openvpn/plugin/lib/openvpn-auth-pam.so'
       $namespecific_rclink = false
+      $default_easyrsa_ver = '3.0'
+      $easyrsa_source      = '/usr/share/easy-rsa/3'
 
-      # Redhat/Centos >= 7.0
-      if(versioncmp($::operatingsystemrelease, '7.0') >= 0) and $::operatingsystem != 'Amazon' {
-        $additional_packages = ['easy-rsa']
-        $ldap_auth_plugin_location = undef
-        $systemd = true
-        $easyrsa_source      = '/usr/share/easy-rsa/3'
-        $default_easyrsa_ver = '3.0'
-      # Redhat/Centos == 6.0
-      } elsif(versioncmp($::operatingsystemrelease, '6.0') >= 0) and $::operatingsystem != 'Amazon' {
-        $additional_packages = ['easy-rsa','openvpn-auth-ldap']
-        $ldap_auth_plugin_location = '/usr/lib64/openvpn/plugin/lib/openvpn-auth-ldap.so'
-        $systemd = false
-        $easyrsa_source      = '/usr/share/easy-rsa/3'
-        $default_easyrsa_ver = '3.0'
-      # Redhat/Centos < 6.0
-      } else {
-        $additional_packages = ['easy-rsa']
-        $ldap_auth_plugin_location = undef
-        $systemd = false
-        $easyrsa_source      = '/usr/share/easy-rsa/2.0'
-        $default_easyrsa_ver = '2.0'
+      case $facts['os']['release']['major'] {
+        '7': {
+          $additional_packages = ['easy-rsa']
+          $ldap_auth_plugin_location = undef
+          $systemd = true
+        }
+        '6': {
+          $additional_packages = ['easy-rsa','openvpn-auth-ldap']
+          $ldap_auth_plugin_location = '/usr/lib64/openvpn/plugin/lib/openvpn-auth-ldap.so'
+          $systemd = false
```

Comment:
suggestion for future enhancements: ```$systemd``` can be set to ```true``` if the fact ```[service_provider```](https://github.com/puppetlabs/puppetlabs-stdlib#service_provider) returns ```systemd```